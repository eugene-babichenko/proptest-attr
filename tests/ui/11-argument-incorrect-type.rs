use proptest::prelude::*;
use proptest_attr::proptest;

fn main() {
    basic_test();
}

#[proptest(strategy = "0..10u8")]
fn basic_test(_value: u32) -> Result<(), TestCaseError> {
    Ok(())
}