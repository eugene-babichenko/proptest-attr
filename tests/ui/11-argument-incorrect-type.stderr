error[E0631]: type mismatch in closure arguments
 --> $DIR/11-argument-incorrect-type.rs:8:1
  |
8 | #[proptest(strategy = "0..10u8")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected signature of `fn(u8) -> _`
9 | fn basic_test(_value: u32) -> Result<(), TestCaseError> {
  | ------------------------------------------------------- found signature of `fn(u32) -> _`
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
